content/source/docs/enterprise/run/index.html.md

```diff
@@ -82,6 +82,27 @@ If you need to temporarily stop runs from being queued, you can lock the workspa
 
 You can find the lock button in the workspace settings page. Locking a workspace requires write or admin access.
 
+## Installing Terraform Providers
+
+### Providers Distributed by HashiCorp
+
+TFE runs `terraform init` before every plan or apply, which automatically downloads any [providers](/docs/configuration/providers.html) Terraform needs.
+
+Private installs of TFE can automatically install providers as long as they can access registry.terraform.io. If that isn't feasible due to security requirements, you can manually install providers. Use [the `terraform-bundle` tool][bundle] to pre-download all the providers you need and bundle them with a version of Terraform, use the admin pages to upload the bundle as a new Terraform version, and configure your workspaces to use that bundled Terraform version.
+
+[bundle]: https://github.com/hashicorp/terraform/tree/master/tools/terraform-bundle#installing-a-bundle-in-on-premises-terraform-enterprise
+
+### Third-Party Providers
+
+-> **Note:** We are investigating how to improve third-party provider installation, so this information might change in the near future.
+
+Currently, there are two ways to use third-party provider plugins with TFE.
+
+- Add the provider binary to the VCS repo (or manually-uploaded configuration version) for any workspace that uses it. You must include the compiled `linux_amd64` version of the plugin, and it must be stored at `terraform.d/plugins/linux_amd64/<PLUGIN NAME>` (as a relative path from the root of the working directory).
+
+    When adding plugins to a repo, you can add them directly or add them as a Git submodule. (Make sure each workspace's ["Include submodules on clone" setting](../workspaces/settings.html#include-submodules-on-clone) is enabled.) Submodules can reduce repo sizes when multiple workspaces use the same third-party provider.
```

Comment:
The first sentence is a little unclear on whether this is one or several submodules. I'd also add "If you choose to them as (submodules|a submodule)," at the beginning of the parenthetical.